components/atom/input/src/Input/index.js

```diff
@@ -1,16 +1,16 @@
 import React from 'react'
 import AddonHoc from './Features/Addon'
+import WithIconHoc from './Features/WithIcon'
 import Component, {inputSizes} from './Component'
 
 const WithAddonComponent = AddonHoc(Component)
+const WithIconComponent = WithIconHoc(Component)
 
 export default props => {
-  const {leftAddon, rightAddon} = props // eslint-disable-line react/prop-types
-  return leftAddon || rightAddon ? (
-    <WithAddonComponent {...props} />
-  ) : (
-    <Component {...props} />
-  )
+  const {leftAddon, leftIcon, rightAddon} = props // eslint-disable-line react/prop-types
+  if (leftAddon || rightAddon) return <WithAddonComponent {...props} />
```

Comment:
Si I can't have an input with icon AND right addon ?
Does it make sense to treat icons as addons as well ?